crates/ra_hir/src/semantics.rs

```diff
@@ -40,6 +41,35 @@ pub enum PathResolution {
     AssocItem(AssocItem),
 }
 
+impl PathResolution {
+    pub fn in_type_ns(self) -> Option<TypeNs> {
```

Comment:
hmm I wouldn't really want to have this public -- `PathResolution` is a 'simplified' type for the facade that loses the namespace information, so trying to infer the namespace again might be lossy. It'll be good enough for this purpose, but might lead to problems otherwise. (Apart from the fact that we don't want to expose `TypeNs`.)